joplin/api/schema.py

```diff
@@ -179,11 +179,62 @@ class ContextualNavData(graphene.ObjectType):
     # related_to = graphene.List(JanisBasePageTopicCollectionNode)
 
 
+'''
+    Note: we do NOT want to use DjangoObjectType for the User model.
+    Otherwise owners and users will be visible on all nodes by default.
+    We want Nodes to be explicit if they want to resolve users,
+    and we want those resolvers to be wrapped in a @superuser_required decorator for authorization.
+    TODO: handle importing of department groups for non-superusers.
+'''
+class OwnerNode(graphene.ObjectType):
+    id = graphene.ID()
+    first_name = graphene.String()
+    last_name = graphene.String()
+    email = graphene.String()
+    is_superuser = graphene.String()
+
+    @superuser_required
+    def resolve_id(self, info):
+        return self.id
+
+    @superuser_required
+    def resolve_first_name(self, info):
+        return self.first_name
+
+    @superuser_required
+    def resolve_last_name(self, info):
+        return self.last_name
+
+    @superuser_required
+    def resolve_email(self, info):
+        return self.email
+
+    @superuser_required
+    def resolve_is_superuser(self, info):
+        return self.is_superuser
+
+
+# Add this method to a PageNode to allow authorized users to see page owner data.
+def resolve_owner_handler(self, info):
+    owner = self.owner
```

Comment:
this is for my own edification, do some of our pages not have owners? or is it just in case so nothing breaks?

Reply:
Some of our pages actually don't have owners. They're legacy pages from early on in Joplin v2 that must have been created programmatically. I can't remember the story of how they got there, but they're there.